src/legacy/core_plugins/data/public/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -49,7 +49,7 @@ interface MappingObject {
 }
 
 export interface StaticIndexPattern {
-  fields: FieldType[];
+  fields: FieldType[] | FieldListInterface;
```

Comment:
This is a sad compromise but I'm not sure what should be done about it. I think this is a case were refactoring with types in mind is more difficult than writing the code in the first place.